### ISSUE TITLE
Requirements to build wheel unsuccessful - The 'sklearn' PyPI package is deprecated

### ISSUE THREAD

**Issue by Attested-paper51** [SPFlow/SPFlow#178]
### What did you expect to happen?

When I do "pip3 install spflow", it gave me a build error ("Getting requirements to build wheel did not run successfully."), when it was supposed to successfully install spflow.

### What actually happened?

When I do "pip3 install spflow", it gave me a build error ("Getting requirements to build wheel did not run successfully.")
Here is the stack trace:
![image](https://github.com/SPFlow/SPFlow/assets/79871733/e751e9db-2336-4548-813e-61475b5cf60b)

### Describe your attempts to resolve the issue

I tried following the instructions listed in the console, but the issue still persisted. Such as installing scikit-learn before installing spflow, and checking the requirements.txt to make sure there's no sklearn. I even changed all "from sklearn" to "from scikit-learn" in the source code, and the issue still persisted.

### Steps to reproduce

in VSCode, go to your main project directory, and run this command: "pip3 install spflow"

### System Information

Python version: 3.12.1
SPFlow version: latest
Operating System: Windows 10

### Installed Python Packages

<details>
  asttokens==2.4.1
  blinker==1.7.0
  certifi==2024.2.2
  charset-normalizer==3.3.2
  click==8.1.7
  colorama==0.4.6
  comm==0.2.1
  contourpy==1.2.0
  cycler==0.12.1
  debugpy==1.8.1
  decorator==5.1.1
  executing==2.0.1
  Flask==3.0.2
  Flask-Cors==4.0.0
  fonttools==4.49.0
  idna==3.6
  iniconfig==2.0.0
  ipykernel==6.29.3
  ipython==8.22.2
  itsdangerous==2.1.2
  jedi==0.19.1
  Jinja2==3.1.3
  joblib==1.4.0
  jupyter_client==8.6.0
  jupyter_core==5.7.1
  kiwisolver==1.4.5
  MarkupSafe==2.1.5
  matplotlib==3.8.3
  matplotlib-inline==0.1.6
  nest-asyncio==1.6.0
  numpy==1.26.4
  packaging==23.2
  pandas==2.2.1
  parso==0.8.3
  pillow==10.2.0
  platformdirs==4.2.0
  pluggy==1.4.0
  prompt-toolkit==3.0.43
  psutil==5.9.8
  psycopg2==2.9.9
  pure-eval==0.2.2
  Pygments==2.17.2
  pyparsing==3.1.1
  pytest==8.0.0
  python-dateutil==2.8.2
  pytz==2024.1
  pywin32==306
  requests==2.31.0
  scikit-learn==1.4.2
  scipy==1.13.0
  seaborn==0.13.2
  six==1.16.0
  stack-data==0.6.3
  threadpoolctl==3.4.0
  tornado==6.4
  traitlets==5.14.1
  tzdata==2024.1
  urllib3==2.2.1
  wcwidth==0.2.13
  Werkzeug==3.0.1
</details>
